lib/config.js

```diff
@@ -107,16 +106,17 @@ function hasRules(options) {
  * @returns {Object} The local config object, or an empty object if there is no local config.
  */
 function getLocalConfig(thisConfig, directory) {
-    const localConfigFiles = thisConfig.findLocalConfigFiles(directory),
-        numFiles = localConfigFiles.length,
-        projectConfigPath = ConfigFile.getFilenameForDirectory(thisConfig.options.cwd);
+
+    const projectConfigPath = ConfigFile.getFilenameForDirectory(thisConfig.options.cwd);
+    const localConfigFiles = thisConfig.findLocalConfigFiles(directory);
     let found,
         config = {},
         rootPath;
 
-    for (let i = 0; i < numFiles; i++) {
-
-        const localConfigFile = localConfigFiles[i];
+    for (const localConfigFile of localConfigFiles) {
+        if (localConfigFile === undefined) {
+            continue;
```

Comment:
I think code will never be reached assuming `findLocalConfigFiles` only yields strings, so it would be better to remove the `if` statement.